TestResultSummaryService/perf/DataManagerAggregate.js

```diff
@@ -4,68 +4,119 @@ const ObjectID = require( 'mongodb' ).ObjectID;
 
 class DataManagerAggregate {
     static aggDataCollect(childBuild) {
-        const benchmarkMetricsCollection = {};
-        let name, variant;
+        let aggRawMetricValues = [];
+        let benchmarkMetricsCollection = {}; // this is a raw data collection under one benchmark name/variant
+        let bname, bvariant;
         if (Array.isArray(childBuild.tests) && childBuild.tests.length > 0 ) {
-            name = childBuild.tests[0].benchmarkName;
-            variant = childBuild.tests[0].benchmarkVariant;
-            for ( let {testData} of childBuild.tests){
-                if ( Array.isArray(testData.metrics) ) {
-                    for ( let {name, value} of testData.metrics ){
-                        benchmarkMetricsCollection[name] = benchmarkMetricsCollection[name] || [];
-                        if ( Array.isArray(value) ) {
-                            benchmarkMetricsCollection[name] = benchmarkMetricsCollection[name].concat(value);
+            for ( let {benchmarkName, benchmarkVariant, testData} of childBuild.tests){
+                if (benchmarkName && benchmarkVariant && testData && Array.isArray(testData.metrics) && testData.metrics.length > 0){
+                    // Example: Jenkins build (including 2 benchmark jobs with 2 iterations) in this order: Benchmark job_1, Benchmark job_2, Benchmark job_1, Benchmark job_2
+                    let i = 0;
```

Comment:
We should make `aggRawMetricValues` a JSON object instead of an array since that change could simplify some the logic here. I don't the current logic seems to be convoluted: 
```
let i = 0;
for (; i < aggRawMetricValues.length; i++) {
}
if (i === aggRawMetricValues.length) {
}
```
We can make test (i.e. benchmark name + variant) as the key for it. For each test, if it doesn't exist in aggRawMetricValues, then add the key and add the raw values. Else, just add the raw values. For adding raw values, we can just have one function since the logic is almost the same.